contrib: test: fix runc build on AMIs

runcom opened the pull request:
Signed-off-by: Antonio Murdaca <runcom@redhat.com>

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/kubernetes-incubator/cri-o/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

https://github.com/kubernetes-incubator/cri-o/pull/1298 changed the setup of the AMI to actually skip runc build by default and that's causing the whole CI to fail with the ami built from that PR. (i.e. https://github.com/kubernetes-incubator/cri-o/pull/1297)

Fix it by reverting the logic so that we *always* build runc except when asked otherwise

@giuseppe @mrunalp @nalind PTAL

**- How I did it**

**- How to verify it**

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

runcom commented:
/test all
/test ami

runcom commented:
we're just interested in fixing the AMI here so other failures are expected.

giuseppe commented on contrib/test/integration/main.yml:
```diff
@@ -30,7 +30,7 @@
     - name: clone build and install runc
       include: "build/runc.yml"
-      when: "{{ not use_system_runc | default(True) | bool}}"
+      when: "{{ not use_system_runc | default(False) | bool}}"
```
can you also change the "not use_system_runc" to "use_system_runc" ?

runcom replied:
Nope, that brings us back to the bug :) we don't want to skip runc build by default, we want to build runc **always**

runcom replied:
I can do:

```
when: "{{ don_t_use_system_runc | default(True) | bool}}"
```
if it's more readable to you

mrunalp replied:
@giuseppe Does the default apply to the task or the use_system_runc variable?

runcom replied:
it applies to the whole expression preceding the pipe once evaluated

wking replied:
I think the problem is that [the `default` filter][1] is only switching on whether its input is `None` or not.  So you could have:

    {{ use_system_runc | default(False) | not | bool }}

except that there seems to be no `not` filter.  I recommend we change this to:

    {{ build_runc | default(True) | boolean }}

and @guiseppe can set `build_runc` false when he wants to use the system `runc`.

[1]: http://docs.ansible.com/ansible/latest/playbooks_filters.html#defaulting-undefined-variables

runcom replied:
> {{ build_runc | default(True) | boolean }}

AMIs are almost built to unblock #1297 - I'll change the filter to the one you suggested in a follow up!
Thanks!

runcom commented:
BTW, it worked now and it builds runc in the AMI job:
```
---

TASK [clone runc source repo] **************************************************
task path: /go/src/github.com/kubernetes-incubator/cri-o/contrib/test/integration/build/runc.yml:3
 [WARNING]: when statements should not include jinja2 templating delimiters
such as {{ }} or {% %}. Found: {{ not use_system_runc | default(False) | bool}}
changed: [localhost] => {
    "after": "9f9c96235cc97674e935002fc3d78361b696a69e", 
    "before": null, 
    "changed": true
}

TASK [build runc] **************************************************************
task path: /go/src/github.com/kubernetes-incubator/cri-o/contrib/test/integration/build/runc.yml:9
 [WARNING]: when statements should not include jinja2 templating delimiters
such as {{ }} or {% %}. Found: {{ not use_system_runc | default(False) | bool}}
changed: [localhost] => {
    "changed": true, 
    "chdir": "/go/src/github.com/opencontainers/runc", 
    "params": {
        "BUILDTAGS": "seccomp selinux"
    }, 
    "target": null
}

STDOUT:
---
go build -buildmode=pie  -ldflags "-X main.gitCommit="9f9c96235cc97674e935002fc3d78361b696a69e" -X main.version=1.0.0-rc4+dev " -tags "seccomp selinux" -o runc .
---

TASK [install runc] ************************************************************
task path: /go/src/github.com/kubernetes-incubator/cri-o/contrib/test/integration/build/runc.yml:14
 [WARNING]: when statements should not include jinja2 templating delimiters
such as {{ }} or {% %}. Found: {{ not use_system_runc | default(False) | bool}}
changed: [localhost] => {
    "changed": true, 
    "chdir": "/go/src/github.com/opencontainers/runc", 
    "params": null, 
    "target": "install"
}

STDOUT:
---
install -D -m0755 runc /usr/local/sbin/runc
---

TASK [link runc] ***************************************************************
task path: /go/src/github.com/kubernetes-incubator/cri-o/contrib/test/integration/build/runc.yml:19
 [WARNING]: when statements should not include jinja2 templating delimiters
such as {{ }} or {% %}. Found: {{ not use_system_runc | default(False) | bool}}
changed: [localhost] => {
    "changed": true, 
    "dest": "/usr/bin/runc", 
    "gid": 0, 
    "group": "root", 
    "mode": "0777", 
    "owner": "root", 
    "secontext": "unconfined_u:object_r:bin_t:s0", 
    "size": 20, 
    "src": "/usr/local/sbin/runc", 
    "state": "link", 
    "uid": 0
}
```

TomSweeneyRedHat commented:
LGTM, but it looks like there's some unhappiness in a number of tests.

runcom commented:
/test ami

runcom commented:
/test ami_fedora

runcom commented:
/test ami_fedora

runcom commented:
/test ami_fedora

runcom commented:
We get some weird ansible failure here:
```
No handlers could be found for logger "dnf"
```
@cevich do you have any clue what's actually happening?

cevich commented:
@runcom Yes, it's an Ansible bug in the logging system that can safely be ignored.

However, if it's only recently appearing, that's because somebody changed the run-time environment - possibly a much worse problem.

This is why I want to move toward a version-locked environment, something like what ``contrib/test/venv-ansible-playbook.sh`` sets up.  Even though that has bugs, at least they're known-bugs, but more importantly,  _we_ are in control of when the runtime environment changes.

mrunalp commented:
/test ami_fedora

runcom commented:
/test ami_fedora

runcom commented:
/test ami_fedora

mrunalp commented:
From the last run:
```
  [91mfailed to pull image: rpc error: code = Unknown desc = Error determining manifest MIME type for docker://busybox:1.26.2-uclibc: Get https://auth.docker.io/token?scope=repository%3Alibrary%2Fbusybox%3Apull&service=registry.docker.io: EOF
    Expected error:
        <*status.statusError | 0xc4203e7170>: {
            Code: 2,
            Message: "Error determining manifest MIME type for docker://busybox:1.26.2-uclibc: Get https://auth.docker.io/token?scope=repository%3Alibrary%2Fbusybox%3Apull&service=registry.docker.io: EOF",
            Details: nil,
        }
        rpc error: code = Unknown desc = Error determining manifest MIME type for docker://busybox:1.26.2-uclibc: Get https://auth.docker.io/token?scope=repository%3Alibrary%2Fbusybox%3Apull&service=registry.docker.io: EOF
    not to have occurred[0m
```

runcom commented:
@mrunalp that looks like an issue with the docker hub itself actually :(

runcom commented:
/test ami_fedora

mrunalp commented:
Merging as the tests passed in #1305 from the amis built from this PR.
